numpy/_core/fromnumeric.py

```diff
@@ -2643,6 +2668,35 @@ def all(a, axis=None, out=None, keepdims=np._NoValue, *, where=np._NoValue):
                                   keepdims=keepdims, where=where)
 
 
+def _cumulative_sum_dispatcher(x, /, *, axis=None, dtype=None, out=None,
+                               include_initial=None):
+    return (x, out)
+
+
+@array_function_dispatch(_cumulative_sum_dispatcher)
+def cumulative_sum(x, /, *, axis=None, dtype=None, out=None,
+                   include_initial=False):
+    """
+    Array API compatible cumulative_sum
+    """
+    if out is not None and include_initial:
+        raise ValueError("not supported")
```

Comment:
The error message should be a bit clearer about what is not supported.

Reply:
Yes, right, still WIP.

Reply:
OK, yes, should have seen that this is a draft!

Reply:
Updated, done!